src/main/java/com/minecolonies/coremod/entity/ai/citizen/composter/EntityAIWorkComposter.java

```diff
@@ -138,22 +142,25 @@ private IAIState getMaterials()
             setDelay(2);
             return getState();
         }
-        if (getOwnBuilding().getCopyOfAllowedItems().isEmpty())
+
+        final List<ItemStorage> list = getOwnBuilding().getModule(GroupedItemListModule.class).map(m -> m.getList(COMPOSTABLE_LIST)).orElse(ImmutableList.of());
+        if (list.isEmpty())
         {
             complain();
             return getState();
         }
```

Comment:
Again, this is not directly the way you should use an Optional.

Reply:
I can't return getState within the optional map or whatever though, so that's the way it is.

Reply:
Yes you can, why would you not?